elisctl/lib/api_client.py

```diff
@@ -77,13 +79,12 @@ def url(self) -> str:
             self._url = f'{_url}{"/v1" if self._use_api_version else ""}'
         return self._url
 
-    def get_token(self) -> str:
+    def get_token(self, max_login_lifetime: Optional[int]) -> str:
         # self.post cannot be used as it is dependent on self.get_token().
-        response = requests.post(
-            f"{self.url}/auth/login",
-            json={"username": self.user, "password": self.password},
-            headers=HEADERS,
-        )
+        login_data = {"username": self.user, "password": self.password}
+        if max_login_lifetime:
+            login_data["max_token_lifetime"] = max_login_lifetime  # type: ignore
```

Comment:
The type can be provided inline for the `login_data` variable.
```suggestion
        login_data: Dict[str, Union[str, int]] = {"username": self.user, "password": self.password}
        if max_login_lifetime:
            login_data["max_token_lifetime"] = max_login_lifetime
```